docs/source/distributed.pipelining.rst

```diff
@@ -0,0 +1,178 @@
+.. role:: hidden
+    :class: hidden-section
+
+Pipeline Parallelism
```

Comment:
After adding this is there a plan to dedup with the Readme added in the first PR? Seems like we wouldn't need that anymore but I'm not sure if the content is 100%same

Reply:
Good point. Nice to reduce maintenance load.